packages/medusa/src/api/routes/admin/tax-rates/update-tax-rate.ts

```diff
@@ -183,8 +182,8 @@ export class AdminPostTaxRatesTaxRateReq {
   region_id?: string
 
   @IsOptional()
-  @IsType([Number, null])
-  rate?: number | null
```

Comment:
thought: the design of taxes includes support for a nullable rate - the idea behind this is that if you are integrating with a third party system you may be in a situation where the numerical rate is dynamically computed through a 3rd party request. In this scenario, you use the TaxRate to tag the tax code to a product, collection, etc., the code is then used to compute and respond with the correct rate. E.g.:

- Product - Book
- TaxRate - code: book, rate: null
- Cart - items: [book]
- Total -> makes call to 3rd party with code: book and address: "some place with hyper local tax rates" -> responds with rate: 0.0932 -> This rate is used in cart total calculation

Reply:
Yes, this doesn't change with this PR. You can still create and update tax rates with a null rate. The changes here correct what is an inconsistency - and recently breaking - way of supporting this. 

I've added tests to ensure, we are aligned on the intended behaviour. Can you confirm?

Reply:
This doesn't change with this PR. You can still create and update tax rates with a null rate. The changes here correct what is an inconsistent - and recently breaking - way of supporting a nullable payload at the API level.

I've added tests to ensure we are aligned on the intended behaviour. Can you confirm?